src/image_generation/image_generator.py

```diff
@@ -55,11 +57,66 @@ def create_image_generator(model_name: str) -> ImageGenerator:
     model_name = model_name.lower()
     if model_name == 'dall-e':
         image_generator = DallEImageGenerator()
+
+    elif model_name == 'stable-diffusion':
+        image_generator = StableDiffusionImageGenerator()
     else:
         raise ValueError(f'Invalid model name: {model_name}')
     return image_generator
 
 
+class StableDiffusionImageGenerator(ImageGenerator):
+
+    def generate_image(self, image_prompt: str, width: int, height: int) -> str:
+        '''
+        Generates an image from a prompt of a certain size
+        Args:
+            image_prompt (str): The prompt to generate the image from. Must be less than 1000 characters.
+            width (int): The width of the image. Must be 256, 512, or 1024
+            height (int): The height of the image. Must be 256, 512, or 1024
+        Returns:
+            The URL of the generated image
+        '''
+        url = "https://stablediffusionapi.com/api/v4/dreambooth"
+
+        payload = json.dumps({
+        "key": Config().STABLE_DIFFUSION_API_KEY,
+        "model_id": "midjourney",
+        "prompt": image_prompt,
+        "negative_prompt": "text",
+        "width": "1024",
+        "height": "1024",
+        "samples": "1",
+        "num_inference_steps": "30",
+        "safety_checker": "no",
+        "enhance_prompt": "yes",
+        "seed": None,
+        "guidance_scale": 7.5,
+        "multi_lingual": "no",
+        "panorama": "no",
+        "self_attention": "no",
+        "upscale": "no",
+        "embeddings_model": None,
+        "lora_model": None,
+        "tomesd": "yes",
+        "use_karras_sigmas": "yes",
+        "vae": None,
+        "lora_strength": None,
+        "scheduler": "UniPCMultistepScheduler",
+        "webhook": None,
+        "track_id": None
+        })
+
+        headers = {
+        'Content-Type': 'application/json'
+        }
+
+        response = requests.request("POST", url, headers=headers, data=payload)
+        sleep(8)
```

Comment:
There should be a better solution to asycrounus python code then to just hope for the sleep to work. I just do not know how xD